lambda/delete_tile_index_entry_lambda.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2016 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Lambda that removes the entry from the DynamoDB tile index for a given chunk.
+This happens after a chunk is successfully placed in the cuboid bucket as
+Boss cuboids.
+
+Does not handle errors.  Assumed that this lambda will be invoked
+asynchronously and caller will rely on AWS' automatic retries (3).
+
+event: {
+    'chunk_key': (str) key identifying entry in tile index.
+    'task_id': (int) Id of the ingest job that the chunk belongs to.
+    'region': (str) AWS region tile index lives in.
+    'tile_index': (str) Name of the DynamoDB table that is the tile index.
+}
+"""
+
+import boto3
+
+
+def handler(event, context):
+    """
+    Entry point for the lambda function.
+
+    Args:
+        event (dict): Lambda input parameters.  See module level comments.
+        context (dict): Standard lambda context parameter (ignored).
+
+    Raises:
+    """
+    resp = delete_entry(event)
+
+def delete_entry(event):
+    chunk_key = event['chunk_key']
+    task_id = event['task_id']
+    key = {'chunk_key': {'S':chunk_key}, 'task_id': {'N': str(task_id)}}
+
+    dynamo = boto3.client('dynamodb', region_name=event['region'])
+    resp = dynamo.delete_item(Key=key, TableName=event['tile_index'])
```

Comment:
The last line of delete_tile_index_entry_lambda.py doesn't check the resp for a 200.  I think it would be helpful to print the error if one occurs.  I don't expect of you to deal with it because will most likely just assume it will be auto deleted after a given time period but it may help for debugging when there are issues.

Reply:
Yeah, the exception will be logged automatically. This is what was logged last week when I was testing my dead letter queue:

```
test dlq: Exception
Traceback (most recent call last):
File "/var/task/delete_tile_index_entry_lambda.py", line 28, in handler
resp = delete_entry(event)
File "/var/task/delete_tile_index_entry_lambda.py", line 31, in delete_entry
raise Exception('test dlq')
Exception: test dlq
```

Reply:
Ok.  So it wont be a 403 if they don't have access but an exception which will be documented.